conda/notices/types.py

```diff
@@ -76,8 +77,15 @@ def _parse_iso_timestamp(iso_timestamp: Optional[str]) -> Optional[datetime]:
 
     @classmethod
     def get_cache_key(cls, url: str, name: str, cache_dir: Path) -> Path:
-        """Returns the place where this channel response will be stored as cache"""
+        """
+        Returns the place where this channel response will be stored as cache
+
+        We use a naive hashing algorithm to avoid issues with filenames (e.g. illegal or
+        problematic characters). We don't care the that this is a weak hash.
+        """
         url_obj = parse.urlparse(url)
-        path = url_obj.path.replace("/", "-")
-        cache_filename = f"{name}{path}"
+        bytes_filename = f"{name}{url_obj.path}".encode()
+        md5_hash = hashlib.sha256(bytes_filename)
```

Comment:
change the variable name to `hash`. it would also be reasonable to truncate the hash.

Reply:
won't truncating the hash lead to possible collisions?

Reply:
There's a nice approximation at https://en.wikipedia.org/wiki/Birthday_attack. 64 bits or 8 bytes (half the length of md5) gives about a one in a million chance of collision, if you've hashed 6.1 million items.

Reply:
I'm curious about the channel name / path instead of the full URL. Conda usually expects channel names to be unique right?

Reply:
Thanks @dholth, this function has been slowly mangled by layers of comments, and I forgot the that U in URL stands for "Unique" 🙈 . I'm going to update this to be more clear and just to hash the URL string instead. No channel name needed!